src/pages/index.js

```diff
@@ -5,77 +5,210 @@ import PopupWithImage from "../components/PopupWithImage.js";
+    authorization: "7b855de8-b589-492b-9e1c-b631d0b0f4c2",
+    "Content-Type": "application/json",
+  },
+});
+
+const cardList = new Section(
   {
-    items: initialCards,
-    renderer: renderCard,
+    renderer: (cardData) => {
+      cardList.addItem(createCard(cardData));
+    },
   },
   ".elements__cards"
 );
 
+api
+  .getInitialCards()
+  .then((res) => {
+    // console.log(">>Initial Cards", cardData);
+    cardList.renderItems(res.reverse());
+    console.log(res);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+    userInfo.setUserAvatar(userData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
 //FUNCTIONS----------------------
 
+function setButtonText(button, text) {
+  button.textContent = text;
+}
+
 function createCard(cardData) {
-  const card = new Card(cardData, "#element-template", () =>
-    handleImageClick(cardData)
+  const card = new Card(
+    cardData,
+    "#element-template",
+    () => handleImageClick(cardData),
+    handleDeleteButton,
+    handleCardLike
   );
   return card.getView();
 }
 
-function renderCard(cardData) {
-  const card = createCard(cardData);
-  cardsSection.addItem(card);
-}
+// function renderCard(cardData) {
+//   const card = createCard(cardData);
+//   section.addItem(card);
+// }
+
+//Image Popup Render
+
+const imagePopup = new PopupWithImage("#modal-preview");
+imagePopup.setEventListeners();
 
 function handleImageClick(cardData) {
   imagePopup.open(cardData.name, cardData.link);
 }
 
-cardsSection.renderItems();
+function handleCardLike(card) {
+  api
+    .likeCard(card._id, card.isLiked())
+    .then(() => {
+      card.handleLikeIcon();
+    })
+    .catch(console.error);
+}
 
-//User Info Render
+function handleProfileEditSubmit({ name, about }) {
+  editProfilePopup.setLoading(true);
+  return api
+    .updateProfileInfo(name, about)
+    .then((userData) => {
+      userInfo.setUserInfo(userData);
+      editProfilePopup.close();
+    })
+    .finally(() => {});
+}
 
-const userInfo = new UserInfo(".profile__name", ".profile__description");
+function handleAddNewCardSubmit(cardData) {
+  // renderCard(cardData, cardsWrap);
+  api
+    .createCard(cardData)
+    .then((res) => {
+      console.log(res);
+      // const card = createCard(cardData);
+      createCard(res);
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+}
+
+//Confirm Delete Popup Render
 
-function handleProfileEditSubmit(userData) {
-  // const userData = editProfilePopup.getInputValues();
-  userInfo.setUserInfo(userData);
-  editProfilePopup.close();
+const confirmDelete = new PopupWithConfirmation("#confirm-modal");
+confirmDelete.setEventListeners();
+
+function handleDeleteButton(card) {
+  console.log("Delete button clicked");
+  confirmDelete.open();
+  console.log(card._id);
+  confirmDelete.setSubmitAction(() => {
+    api
```

Comment:
Needs correction: you should also notify the user about loading process by changing the button text